### ISSUE TITLE
how to reinitialize carousel,

### ISSUE THREAD

**Issue by raghvpip3** [vitalii-andriiovskyi/ngx-owl-carousel-o#130]
Use case: im loading carousel dynamic, once its purpose is over i have to load new carousel,  how do i update/refresh carousel should start from beginning...

**Comment by ajeetshah** [comment]
I am also looking to reinitialize (or destroy and start) the carousel dynamically. But I didn't find a way to do this in the docs.

**Comment by vitalii-andriiovskyi** [comment]
You can't reinitialize carousel directly. However, the carousel always rerenders (reinitializes) after the array with data for slides changes or the event `window:resize` fires.

@raghvpip3 just update array with data for slides. Should work.

**Comment by AlaaEl-DinAhmed** [comment]
is the carousel reinitialized when the options changed?

**Comment by vitalii-andriiovskyi** [comment]
@ajeetshah 
> is the carousel reinitialized when the options changed?  

No.

**Comment by ajeetshah** [comment]
@AlaaEl-DinAhmed 

> @ajeetshah
> 
> > is the carousel reinitialized when the options changed?
> 
> No.

**Comment by AlaaEl-DinAhmed** [comment]
so it must be rerendered to listen to the new options, am I right?

**Comment by AlaaEl-DinAhmed** [comment]
> @ajeetshah
> 
> > is the carousel reinitialized when the options changed?
> 
> No.

so it must be rerendered to listen to the new options, am I right?

**Comment by vitalii-andriiovskyi** [comment]
@AlaaEl-DinAhmed  
It doesn't listen to the new options. It just listens to the number of `<ng-template carouselSlide>` and event `resize` attached to `window`

**Comment by chiranshu14** [comment]
@vitalii-andriiovskyi After I add more slides dynamically (once it reaches the end) it reinitializes and goes back to slide 0. Can we in any way avoid doing that and just append new slides seamlessly?

**Comment by vitalii-andriiovskyi** [comment]
@chiranshu14 

You have to update the option `startPosition` in your `carouselOptions` before adding slides dynamicly by setting it to the value that `startPosition` from `SlidesOutputData` has.
Adding slides dynamicly always rerenders the carousel, and current options take place in this.

Simple example: 
```typescript
customOptions: OwlOptions = {
    loop: true,
    // ......
}

getPassedData(data: SlidesOutputData) { // <- method that listens to the event `translated`
    this.activeSlides = data;
    console.log('HomeComponent');
    console.log(this.activeSlides);
    if (this.activeSlides.startPosition >= this.carouselData.length - 3) {
      this.customOptions.startPosition = this.activeSlides.startPosition;  // <- update `startPosition`
      this.addData();  // <- method that adds slides data
    }
  }
```

**Comment by chiranshu14** [comment]
@vitalii-andriiovskyi  Thanks for your quick response.

I have implemented a similar solution as suggested by you. I thought there might be a better way to get this done.

I am using the (changed) event listener instead of  (translated) as suggested by you. Does that make any difference? changed seems to work well too...

Thank you!

**Comment by vitalii-andriiovskyi** [comment]
@chiranshu14 

No. It doesn't make difference. The event `changed` fires before the event `translated`. The event `translated` is fired when the transition of carousel is done. The event `changed` fires when new data about new active slides are available. It doesn't wait until the carousel finish transition/swiping left/right. Both events emits the same data.
